tools/util/include/cutlass/util/reference/device/tensor_foreach.h

```diff
@@ -133,7 +134,8 @@ struct BlockForEach {
 #if defined (CUTLASS_ENABLE_SYCL)
       // TODO: query the queue for block size
       block_size = 128;
-      grid_size = cute::ceil_div(capacity, block_size);
+      // Ensure global range doesn't overflow int
+      grid_size = std::min(capacity, static_cast<size_t>(std::numeric_limits<int>::max())) / block_size;
```

Comment:
Did you consider using something like `sycl::info::device::max_work_group_size` or `sycl::info::device::max_work_item_dimensions`?